cmd/snap-confine/ns-support.c

```diff
@@ -238,8 +239,76 @@ void sc_close_ns_group(struct sc_ns_group *group)
 	free(group);
 }
 
+static dev_t find_base_snap_device(const char *base_snap_name,
+				   const char *base_snap_rev)
+{
+	// Find the backing device of the base snap.
+	// TODO: add support for "try mode" base snaps that also need
```

Comment:
Hm, if `try mode` does not work with this function and we die() below on not-found, isn't that dangerous? I.e. should we just keep going if we cannot find the base snaps dev_t?

Reply:
It will not die but it will find the block device for the (typically) home partition. What I mean by this is that we want to remember which directory of that block device was used so that when you do this, it still detects the staleness:

```
cd ~/foo/base
snap try
cd ~/bar/base
snap try
```

Reply:
Thanks for explaining this.